app/shared/components/paper-component/PaperComponent.tsx

```diff
@@ -0,0 +1,33 @@
+import { Box, Paper, SxProps, Theme } from '@mui/material';
+import { ReactNode } from 'react';
+
+import { mainHexPallete } from '../design-system/all-components/theme/colors';
+import { styles } from './PaperComponent.styles';
+import { ResponsiveStyle } from '~/types/types/common.types';
+
+import { sxToArray } from '~/lib/utils/sxToArray';
+
+type PaperComponentProps = Readonly<{
+  children: ReactNode;
+  padding: ResponsiveStyle;
+  backgroundColor: (typeof mainHexPallete)[keyof typeof mainHexPallete] | string;
+  sx?: SxProps<Theme>;
+}>;
+
+export default function PaperComponent({ children, sx, padding, backgroundColor }: PaperComponentProps) {
```

Comment:
Usually, background color and padding are passed through styles, so separate props are redundant. Moreover, they are mandatory, which obliges us to use them in every call.

In my opinion, it would be better to abandon our own type altogether and use the props of the Paper component from the Mui library:
```suggestion
interface PaperComponentProps extends PaperProps {
  childrenSx: SxProps<Theme>;
}

export default function PaperComponent({ children, sx, childrenSx, elevation = 0, ...props }: PaperComponentProps ) {
```

Then we can simply pass props to the Paper component, but add custom styles.

This way, we will get rid of unnecessary types and inline styles.

Reply:
> Usually, background color and padding are passed through styles, so separate props are redundant. Moreover, they are mandatory, which obliges us to use them in every call.
> 
> In my opinion, it would be better to abandon our own type altogether and use the props of the Paper component from the Mui library:
> 
> Then we can simply pass props to the Paper component, but add custom styles.
> 
> This way, we will get rid of unnecessary types and inline styles.

padding and backgroundColor are always required by design, which keeps the component consistent with the design system

Reply:
Yes, but you still pass them inline styles later. It's better to pass them in the style object right away. Also, there might be a need for a default style somewhere, then you don't need to pass them at all, and with the current implementation that's impossible.